### PR TITLE
Compliance: Improve tmp handling

### PR DESCRIPTION
**Pull request by vkrizan**
RHICOMPL-1196

### All Pull Requests:

* [X] Have you followed the guidelines in our Contributing document?
* [ ] Is this PR to correct an issue?
* [X] Is this PR an enhancement?

### Enhancement:

Reuse archive's temp dir for the oscap results.
And use tempfile for tailoring file.

### CODE REVIEW

**Comment by vkrizan** [comment]
@akofink @Victoremepunto please review. Thanks.

File: insights/client/apps/compliance/__init__.py [hunk 1]
```diff
@@ -32,13 +34,18 @@ def oscap_scan(self):
...
             self.run_scan(
                 profile['attributes']['ref_id'],
                 self.find_scap_policy(profile['attributes']['ref_id']),
-                '/var/tmp/oscap_results-{0}.xml'.format(profile['attributes']['ref_id']),
-                tailoring_file_path=self.download_tailoring_file(profile)
+                self._results_file(archive_dir, profile),
+                tailoring_file_path=tailoring_file
             )
+            if tailoring_file:
+                os.unlink(tailoring_file)
```
Comment (akofink) [inline insights/client/apps/compliance/__init__.py:48]: `unlink` and `remove` seem identical? The [os.unlink docs](https://docs.python.org/3/library/os.html#os.unlink) state:

>This function is semantically identical to remove(); the unlink name is its traditional Unix name.

Nit: IMO `remove` makes it more readable for those who aren't as familiar with Python (or UNIX).

**Review by akofink — APPROVED** [review]
Works great! Thanks, @vkrizan!

File: insights/client/apps/compliance/__init__.py [hunk 2]
```diff
@@ -51,7 +58,11 @@ def download_tailoring_file(self, profile):
         logger.debug(
             "Policy {0} is a tailored policy. Starting tailoring file download...".format(profile['attributes']['ref_id'])
         )
-        tailoring_file_path = "/var/tmp/oscap_tailoring_file-{0}.xml".format(profile['attributes']['ref_id'])
+        tailoring_file_path = tempfile.mkstemp(
+            prefix='oscap_tailoring_file-{0}.'.format(profile['attributes']['ref_id']),
+            suffix='.xml',
+            dir='/var/tmp'
+        )[1]
```
[the remark this answers is not in the document]
Comment (vkrizan) [reply insights/client/apps/compliance/__init__.py:65]: It is almost the same as before, but with the advantages that the `mkstemp` brings (adds a random bit and ensures correct permissions).

**Comment by vkrizan** [comment]
Hi @gravitypriest, mind having a look. Thanks!

**Review by gravitypriest — APPROVED** [review]

**Comment by vkrizan** [comment]
@gravitypriest is there something required from us to have this merged?

**Pull request merged**